src/util/http/HttpClient.cpp

```diff
@@ -211,3 +227,36 @@ HttpOrHttpsResponse sendHttpOrHttpsRequest(
     return sendRequest.operator()<HttpsClient>();
   }
 }
+
+// ____________________________________________________________________________
+template <typename StreamType>
+cppcoro::generator<std::string> HttpClientImpl<StreamType>::readWebSocketStream(
+    std::unique_ptr<HttpClientImpl<StreamType>> client, std::string_view host,
+    std::string_view target) {
+  AD_CORRECTNESS_CHECK(client->stream_);
+  client->ws_ = std::make_unique<boost::beast::websocket::stream<StreamType>>(
+      std::move(*(client->stream_)));
+  client->ws_->handshake(host, target);
+
+  beast::flat_buffer buffer;
+  for (;;) {
+    client->ws_->read(buffer);
+    co_yield beast::buffers_to_string(buffer.data());
+    buffer.consume(buffer.size());
+  }
```

Comment:
[See here for something I found about cancellation ](https://stackoverflow.com/questions/41509996/how-to-cancel-a-boostasioread-operation-while-its-waiting)

This should document, that even normal exits will throw an exception, so that calling code has to handle everything.

Reply:
Or you use an explicit `strand` that you run in this thread etc, but I'll have to llok into that again.